sections/rich-text.liquid

```diff
@@ -20,19 +20,33 @@
 
 <div class="isolate{% unless section.settings.full_width %} page-width{% endunless %}">
   <div class="rich-text content-container color-{{ section.settings.color_scheme }} gradient{% if section.settings.full_width %} rich-text--full-width content-container--full-width{% endif %} section-{{ section.id }}-padding">
-    <div class="rich-text__blocks">
-      {%- for block in section.blocks -%}
-        {%- case block.type -%}
-          {%- when 'heading' -%}
-            <h2 class="{{ block.settings.heading_size }}" {{ block.shopify_attributes }}>{{ block.settings.heading | escape }}</h2>
-          {%- when 'text' -%}
-            <div class="rich-text__text rte" {{ block.shopify_attributes }}>{{ block.settings.text }}</div>
-          {%- when 'button' -%}
-            <a{% if block.settings.button_link == blank %} role="link" aria-disabled="true"{% else %} href="{{ block.settings.button_link }}"{% endif %} class="button{% if block.settings.button_style_secondary %} button--secondary{% else %} button--primary{% endif %}" {{ block.shopify_attributes }}>
-              {{ block.settings.button_label | escape }}
-            </a>
-        {%- endcase -%}
-      {%- endfor -%}
+    <div class="rich-text__wrapper rich-text__wrapper--{{ section.settings.desktop_content_position }}{% if section.settings.full_width %} page-width{% endif %}">
+      <div class="rich-text__blocks {{ section.settings.content_alignment }}">
+        {%- for block in section.blocks -%}
+          {%- case block.type -%}
+            {%- when 'heading' -%}
+              <h2 class="rich-text__heading rte {{ block.settings.heading_size }}" {{ block.shopify_attributes }}>
+                {{ block.settings.heading | replace: 'p>', 'span>' }}
+              </h2>
+            {%- when 'caption' -%}
+              <p class="rich-text__caption {{ block.settings.text_style }} {{ block.settings.text_style }}--{{ block.settings.text_size }} {{ block.settings.text_style }}" {{ block.shopify_attributes }}>
+                {{ block.settings.caption | escape }}
+              </p>
+            {%- when 'text' -%}
+              <div class="rich-text__text rte" {{ block.shopify_attributes }}>
+                {{ block.settings.text }}
+              </div>
+            {%- when 'button' -%}
+              {%- if block.settings.button_label != blank -%}
+                <div class="rich-text__button">
+                  <a{% if block.settings.button_link == blank %} role="link" aria-disabled="true"{% else %} href="{{ block.settings.button_link }}"{% endif %} class="button{% if block.settings.button_style_secondary %} button--secondary{% else %} button--primary{% endif %}" {{ block.shopify_attributes }}>
+                    {{ block.settings.button_label | escape }}
+                  </a>
+                </div>
```

Comment:
Thanks Ludo, I think that's a great suggestion!

Here are my thoughts:

- I thought about **not removing** the `limit` for the button block but decided against it since the other blocks' `limit` were removed and because it was a design decision. I thought that having multiple button blocks _could_ cause confusion or increase cognitive load for buyers (and potentially merchants).
  - My mental model for most theme sections is the association of one button per section.
- Re: **inline** or **stacked**: I'm not opposed to having them inline and the Image banner section does for its buttons. I decided to stack them since the other blocks (heading and text) behave that way. I know, probably not a good enough reason but that leads to the next point.
- Perhaps we could introduce a new setting for this new pattern (when removing the block limit of buttons in a section), a toggle/checkbox setting to choose between "Inline" and "Stacked" which could add even more flexibility 🤔 ?

In either case + thinking about this more, this definitely feels like something we could explore further with UX and if its decided that we should implement the multiple buttons differently (different than what's in this PR), then we can handle it in a follow-up.

What are your thoughts?

Reply:
Yeah I think a follow up exploration/discussion with designers make sense to see if having an option for those different layouts makes sense or not. 

As for now, my feeling is that the behaviour should be similar to what it is in other sections (like image with text)  for continuity. We can confirm with Yoann what he thinks is best. 👍

Reply:
Hello! I like the idea to have them on the same row like we have in Image banner. 
- Since there is no limit to add button blocks on the editor, maybe we can have them on the same row as long as they are one after another. For example, if after a series of button blocks i use a heading blocl, then heading would display below. But if i add a series of buttons blocks again after, they also would be on the same row again.
- If we do that, maybe buttons on the same row should have the same width as it is currently in image banner? Just so we are consistent.
- I assume that if we have a series of buttons on the same row, there would be on two rows after a certain width. Since there is a max width for heading and description. @andrewetchen @ludoboludo let me know if it doesn't make sense.